spec/models/reservations_validations_spec.rb

```diff
@@ -24,4 +24,24 @@
     reservation.update_attributes reserve_start_at: now, reserve_end_at: now + 1.hour + 5.minutes
     expect(reservation.errors[:base]).to be_present
   end
+
+  describe "starts_before_cutoff" do
+    context "when there is a cutoff" do
+      before :each do
+        reservation.product.update_attribute(:cutoff_time, 2)
+      end
+
+      context "when reservation is after the cutoff" do
+        before { reservation.assign_attributes(reserve_start_at: Time.now + 3.hours, reserve_end_at: Time.now + 4.hours) }
```

Comment:
If you set `reserve_start_at` using the `reservation` factory, it will automatically set `reserve_end_at` to 1 hour later. Something like `FactoryGirl.build(:setup_reservation, reserve_start_at: 3.hours.from_now)`.